cc-cli/Program.cs

```diff
@@ -67,39 +69,30 @@ static void Main(string[] args)
                 _logger.Log(argHandler.ArgData.ToString(), MessageType.DebugInfo);
             }
 
-            // Setup Local Encrypted Storage Config
-            var localStorageConfig = new LocalStorageConfiguration()
-            {
-                AutoLoad = true,
-                AutoSave = true,
-                EnableEncryption = true,
-                EncryptionSalt = Convert.ToBase64String(
-                    Encoding.ASCII.GetBytes(
-                        provider.GetRequiredService<IConfiguration>()["StorageSaltString"]
-                    )
-                )
-            };
-            
-            // Create a Local Storage object for Global persisting settings
-            _localStorageGlobalSettings = new LocalStorage(
-                localStorageConfig, 
-                provider.GetRequiredService<IConfiguration>()["StoragePassword"]
-            );
+            LocalStorage localStorageGlobalSettings = null;
 
-            var storageKeyBase = "cc-cli:global.";
-            if (_localStorageGlobalSettings.Exists(storageKeyBase + CHECKFORUPDATES))
+            if(!string.IsNullOrEmpty(provider.GetRequiredService<IConfiguration>()["StorageSaltString"])
+                && !string.IsNullOrEmpty(provider.GetRequiredService<IConfiguration>()["StoragePassword"]))
             {
-                _checkForUpdates = (bool)_localStorageGlobalSettings.Get(storageKeyBase + CHECKFORUPDATES);
+                localStorageGlobalSettings = SetupEncryptedLocalStorage(
+                    provider.GetRequiredService<IConfiguration>()["StorageSaltString"],
+                    provider.GetRequiredService<IConfiguration>()["StoragePassword"]
+                );
             }
             else
             {
-                _localStorageGlobalSettings.Store(storageKeyBase + CHECKFORUPDATES, _checkForUpdates);
-                _localStorageGlobalSettings.Persist();
+                _logger.Log($"One or more of the encryption secrets are invalid or missing.", MessageType.Warning);
+                _logger.Log($"Do you want to continue using an unencrypted local storage?", MessageType.DisplayText);
```

Comment:
Does the user know the risks of using unencrypted storage? Is this not recommended? If so, should we state that?

Reply:
We could add an update the README to mention what we are saving in Local Storage.
Personally I believe the risk to be low if at all.
This is also an error that should never happen for our cli, and is more of an example for OEMs or other devs who use our code as an example.

Reply:
Ok-- I trust your judgment. If it would add value to describe the contents of the Local Storage in the ReadMe then I support that. But, if you think it would be ignored by 99% of users, then let's just leave it out until someone asks.

Reply:
I am not too worried about unencrypted local storage. Those credentials are transient and can be pretty easily revoked/cycled if necessary.
It is not ideal to have things like this lying around in unencrypted storage, and if we start to see a high number of IP addresses using the same tokens then we should pay attention (they may be sharing the file around willy-nilly) but that is a low probability event.